plugins/woocommerce-admin/client/products/hooks/use-product-attributes.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import {
+	EXPERIMENTAL_PRODUCT_ATTRIBUTE_TERMS_STORE_NAME,
+	ProductAttribute,
+	ProductAttributeTerm,
+} from '@woocommerce/data';
+import { resolveSelect } from '@wordpress/data';
+import { useCallback, useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { sift } from '../../utils';
+
+type useProductAttributesProps = {
+	allAttributes: ProductAttribute[];
+	isVariationAttributes?: boolean;
+	onChange: ( attributes: ProductAttribute[] ) => void;
+	productId?: number;
+};
+
+export type EnhancedProductAttribute = ProductAttribute & {
```

Comment:
I renamed this from `HydratedProductAttribute` to `EnhancedProductAttribute` as I don't think this was being "hydrated" as much as we are appending new data to the product attribute.